lib/mixlib/install/product.rb

```diff
@@ -20,17 +20,21 @@
 module Mixlib
   class Install
     class Product
-      def initialize(&block)
+      def initialize(key, &block)
+        @product_key = key
         instance_eval(&block)
       end
 
       DSL_PROPERTIES = [
         :config_file,
         :ctl_command,
+        :product_key,
```

Comment:
Not sure why we are adding `product_key` when `product` maps to the product keys. It looks like we can just use `product` and  override where necessary (downloads_url or github_repo). If I'll take another look, but that is my impression.